src/inmanta/data/__init__.py

```diff
@@ -5442,6 +5442,22 @@ class EnvironmentMetricsTimer(BaseDocument):
     __primary_key__ = ("environment", "metric_name", "timestamp")
 
 
+class DiscoveredResources(BaseDocument):
+    """
+    discovered resources
+
+    :param environment: the environment of those resources
+    :param discovered_resource_name: The name of the resources
+    :param values: The values
+    """
+
+    environment: uuid.UUID
+    discovered_resource_name: str
+    values: list[object]
+
+    __primary_key__ = ("environment", "discovered_resource_name", "values")
```

Comment:
We probably need more than just the name if we want to avoid conflicts between different providers. Unless we use a fully qualified name or the provider's name (because that one is already checked for uniqueness irrc). But even then I think we need to at least be able to differentiate the (potential) agent that discovered it because it's part of a resource's identity, right?

Edge cases we should support (or at least handle gracefully) I can think of off the top of my head:
- two modules that have nothing to do with each other that happen to use the same name for one of their resources
- handlers like the ones for `std::Package`: multiple handler types for a single resource type
- multiple agents discover resources that happen to have the same attributes, e.g. the same vlan interface on different hosts

Reply:
@sanderr i'm not sure how to handle the 2 first edge cases. Do you have an Idea?
for the third one I will store the UUID of the agent that discovered the resource in the same record as the resource.

Reply:
We decided to store them by ResourceID, so that would include the name of the agent

Reply:
Perfect.